src/modules/tracker.js

```diff
@@ -2235,6 +2235,390 @@ class Tracker {
+        section,
+        intent_result_id: intentResultId,
+        search_result_id: searchResultId,
+        num_results_viewed: numResultsViewed,
+        items: items && Array.isArray(items) && items.slice(0, 100).map((item) => helpers.toSnakeCaseKeys(item, false)),
+        intent,
+      };
+
+      const requestURL = `${requestPath}${applyParamsAsString({}, this.options)}`;
+      const requestMethod = 'POST';
+      const requestBody = applyParams(bodyParams, { ...this.options, requestMethod });
+
+      this.requests.queue(
+        requestURL,
+        requestMethod,
+        requestBody,
+        networkParameters,
+      );
+      this.requests.send();
+
+      return true;
+    }
+
+    this.requests.send();
+
+    return new Error('parameters are required of type object');
+  }
+
+  /**
+   * Send ASA search submitted event
+   *
+   * @function trackAssistantSearchSubmit
+   * @param {object} parameters - Additional parameters to be sent with request
+   * @param {string} parameters.intent - Intent of user request
+   * @param {string} parameters.searchTerm - Term of submitted assistant search event
+   * @param {string} parameters.userInput - The current assistant search query
+   * @param {string} parameters.searchResultId - resultId of search result the clicked item belongs to
+   * @param {string} [parameters.section] - The section name for the item Ex. "Products"
+   * @param {string} [parameters.groupId] - Group identifier of selected item
+   * @param {string} [parameters.intentResultId] - intentResultId from the ASA response
+   * @param {object} [networkParameters] - Parameters relevant to the network request
+   * @param {number} [networkParameters.timeout] - Request timeout (in milliseconds)
+   * @returns {(true|Error)}
+   * @description User submitted an alternative assistant search result search term
+   * @example
+   * constructorio.tracker.trackAssistantSearchSubmit({
+   *     {
+   *         searchResultId: '019927c2-f955-4020-8b8d-6b21b93cb5a2',
+   *         intent: 'show me a recipe for a cookie',
+   *         userInput: 'flour',
+   *         searchTerm: 'flour',
+   *     },
+   * );
+   */
+  trackAssistantSearchSubmit(parameters, networkParameters = {}) {
+    // Ensure parameters are provided (required)
+    if (parameters && typeof parameters === 'object' && !Array.isArray(parameters)) {
+      // Ensure parameters are provided (required)
+      const baseUrl = `${this.behavioralV2Url}assistant_search_submit?`;
+      const {
+        section,
+        intent,
+        searchTerm,
+        userInput,
+        searchResultId,
+        groupId,
+        intentResultId,
+      } = parameters;
+
+      const bodyParams = {
+        intent,
+        section,
+        search_term: searchTerm,
+        user_input: userInput,
+        search_result_id: searchResultId,
+        intent_result_id: intentResultId,
+      };
+
+      if (groupId) {
+        bodyParams.filters = { group_id: groupId };
+      }
```

Comment:
I see you have filters here. Are we only going to support groupId and no other filters??

Reply:
I believe so. This event is similar to trackSearchSubmit so I kept the parameters the same across them.